infra/bridge/src/scripts/20190730-backfill-twitter-profile-data.js

```diff
@@ -0,0 +1,154 @@
+    .toString()
+    .padStart(2, '0')}:${Math.floor(seconds % 60)
+    .toString()
+    .padStart(2, '0')}`
+
+const lookupUsers = screenNames => {
+  return new Promise((resolve, reject) => {
+    client.get(
+      'https://api.twitter.com/1.1/users/lookup.json?screen_name=' +
+        encodeURIComponent(screenNames.join(',')),
+      oAuthToken,
+      oAuthTokenSecret,
+      (error, response) => {
+        if (error) {
+          reject(error)
+        } else {
+          resolve(JSON.parse(response))
+        }
+      }
+    )
+  })
+}
+
+;(async () => {
+  logger.info('Starting backfill script...')
+
+  const startTime = Date.now()
+
+  const attestations = await getAttestationsWithoutProfile()
+  const iterations = Math.floor(attestations.length / USERS_PER_REQUEST) + 1
+
+  let i = 0
+
+  let successCounter = 0
+
+  do {
+    logger.info(
+      `Estimated time to complete: ${getFormattedTime(3 * (iterations - i))}`
+    )
+
+    const startIndex = i * USERS_PER_REQUEST
+    const screenNames = attestations
+      .slice(startIndex, startIndex + USERS_PER_REQUEST)
+      .map(attestation => {
+        if (attestation.username) {
+          return attestation.username
+        }
+
+        // For old twitter attestations, we will have the screen name in `value` field and `username` will be null
+        // https://github.com/OriginProtocol/origin/commit/00f25f5fda75674c7888963496da6d641a0f05db#diff-8b4e2172b733ab5d1803f67ee70f51e6R38-R42
+        return attestation.value
+      })
+
+    if (screenNames.length === 0) {
+      // We are done
+      break
+    }
+
+    const txn = await db.sequelize.transaction()
+
+    try {
+      const response = await lookupUsers(screenNames)
+
+      response.forEach(async user => {
+        await db.Attestation.update(
+          {
+            value: user.id,
+            username: user.screen_name,
+            profileUrl: `https://twitter.com/${user.screen_name}`,
+            profileData: user
+          },
+          {
+            where: {
+              [db.Sequelize.Op.or]: [
+                {
+                  value: user.screen_name,
+                  username: null
+                },
+                {
+                  username: user.screen_name
+                }
+              ]
+            }
+          }
+        )
+      })
+
+      logger.info(`Fetched profile info for ${screenNames.length} users`)
+
+      await txn.commit()
+      logger.info(`Committed to DB`)
+
+      successCounter = successCounter + screenNames.length
+    } catch (error) {
+      await txn.rollback()
+      logger.error(`Error while fetching user data`, error)
+    }
+
+    i++
+  } while (i < iterations && (await waitFor(API_INTERVAL)))
+
+  const endTime = Date.now()
+
+  logger.info()
+  logger.info(
+    `Total time taken: ${getFormattedTime((endTime - startTime) / 1000)}`
+  )
+  logger.info(`Fetched ${successCounter} profiles from Twitter`)
+
+  // Wait for DB commits to complete
```

Comment:
nit: since you use await before DB calls, I don't think that's true ?

Reply:
It's weird but the thing is the script was getting killed before the commit completes. I actually had `process.exit(0)` in this place. The process takes a couple of seconds after the DB call promise has resolved. So, `process.exit(0)` was killing it prematurely. Might be an issue with `sequelize` plugin.

Reply:
I think it's because the main function defined at line 85 is async.
One possible solution:
 - define a main async function
 - call process.exit within a .then callback
Example:

async main() {
  ... do async stuff
}

main()
  .then(() => {
    logger.info('Main done')
    proces.exit()
  })
  .catch(e => {
   logger.error('Failure', e)
   process.exit(1)
  })

Reply:
See an example here: https://github.com/OriginProtocol/origin/blob/master/infra/growth/src/scripts/verifyEvents.js#L129

Reply:
Done